File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-35.cs
public class Solution {
    public int CharacterReplacement(string s, int k) {
        var len = s.Length;
        
        var maxF = 0;
        var l = 0;
        var count = new Dictionary<char, int>();
        var res = 0;
        for (int r = 0; r < len; r++) {
            count[s[r]] = count.GetValueOrDefault(s[r], 0) + 1;
            maxF = Math.Max(maxF, count[s[r]]);
            while (r - l + 1 - maxF > k) {
                count[s[l]]--;
                l++;
            }
            res = Math.Max(res, r - l + 1);
        }
        return res;
    }
}

File: Data Structures & Algorithms/eating-bananas/submission-35.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)
        res = r

        while l <= r:
            totalTime = 0
            k = (l + r) // 2
            for p in piles:
                totalTime += math.ceil(p / k)
            if totalTime <= h:
                res = k
                r = k - 1
            else:
                l = k + 1
        return res
